ksql-engine/src/test/java/io/confluent/ksql/KsqlEngineTest.java

```diff
@@ -313,11 +406,51 @@ public void shouldInferSchemaIfNotPresent() throws Exception {
     assertThat(source.getSqlExpression(), containsString("(FIELD INTEGER)"));
   }
 
+  @Test
+  public void shouldFailIfAvroSchemaNotEvolvable() {
+    // Given:
+    givenTopicWithSchema("T", Schema.create(Type.INT));
+
+    expectedException.expect(KsqlException.class);
+    expectedException.expectMessage(
+        "Cannot register avro schema for T as the schema registry rejected it, "
+            + "(maybe schema evolution issues?)");
+
+    // When:
+    ksqlEngine.execute(
+        "CREATE TABLE T WITH(VALUE_FORMAT='AVRO') AS SELECT * FROM TEST2;",
+        ksqlConfig,
+        Collections.emptyMap());
+
+    // Then:
+    assertThat(metaStore.getSource("T"), is(nullValue()));
```

Comment:
Does this line actually get executed? i.e., won't the previous statement throw an exception?

Reply:
done